SalesforceSDK/Analytics/Model/InstrumentationEvent.cs

```diff
@@ -0,0 +1,237 @@
+        private long _sessionStartTime;
+        private JObject _page;
+        private JObject _previousPage;
+        private JObject _marks;
+
+        internal InstrumentationEvent(string eventId, long startTime, long endTime, string name,
+                         JObject attributes, string sessionId, int sequenceId,
+                         string senderId, JObject senderContext,
+                         SchemaType schemaType, EventType eventType, ErrorType errorType,
+                         DeviceAppAttributes deviceAppAttributes, string connectionType,
+                         string senderParentId, long sessionStartTime, JObject page,
+                         JObject previousPage, JObject marks)
+        {
+            _eventId = eventId;
+            _startTime = startTime;
+            _endTime = endTime;
+            _name = name;
+            _attributes = attributes;
+            _sessionId = sessionId;
+            _sequenceId = sequenceId;
+            _senderContext = senderContext;
+            _schemaType = schemaType;
+            _eventType = eventType;
+            _errorType = errorType;
+            _deviceAppAttributes = deviceAppAttributes;
+            _connectionType = connectionType;
+            _senderParentId = senderParentId;
+            _sessionStartTime = sessionStartTime;
+            _page = page;
+            _previousPage = previousPage;
+            _marks = marks;
+
+        }
+        [JsonProperty]
+        public string EventId => _eventId;
+        [JsonProperty]
+        public long StartTime => _startTime;
+        [JsonProperty]
+        public long EndTime => _endTime;
+        [JsonProperty]
+        public string Name => _name;
+        [JsonProperty]
+        public JObject Attributes => _attributes;
+        [JsonProperty]
+        public string SessionId => _sessionId;
+        [JsonProperty]
+        public int SequenceId => _sequenceId;
+        [JsonProperty]
+        public string SenderId => _senderId;
+        [JsonProperty]
+        public JObject SenderContext => _senderContext;
+        [JsonProperty]
+        public SchemaType Schema => _schemaType;
+        [JsonProperty]
+        public EventType Event => _eventType;
+        [JsonProperty]
+        public ErrorType Error => _errorType;
+        [JsonProperty]
+        public string ConnectionType => _connectionType;
+        [JsonProperty]
+        public DeviceAppAttributes DeviceAppAttributes => _deviceAppAttributes;
+        [JsonProperty]
+        public string SenderParentId => _senderParentId;
+        [JsonProperty]
+        public long SessionStartTime => _sessionStartTime;
+        [JsonProperty]
+        public JObject Page => _page;
+        [JsonProperty]
+        public JObject PreviousPage => _previousPage;
+        [JsonProperty]
+        public JObject Marks => _marks;
+
+        public InstrumentationEvent(JObject json)
+        {
+            json?.ToObject<InstrumentationEvent>();
```

Comment:
did you mean to set some field to this?  Currently this line doesn't do anything